identity_core/examples/merkle_tree.rs

```diff
@@ -0,0 +1,79 @@
+//! cargo run --example merkle_tree
+
+use digest::Digest;
+use identity_core::crypto::merkle_tree::DigestExt;
+use identity_core::crypto::merkle_tree::Hash;
+use identity_core::crypto::merkle_tree::MTree;
+use identity_core::crypto::merkle_tree::Proof;
+use identity_core::crypto::KeyPair;
+use identity_core::error::Result;
+use identity_core::proof::JcsEd25519Signature2020;
+use rand::rngs::OsRng;
+use rand::Rng;
+use sha2::Sha256;
+use std::time::Instant;
+
+const LEAVES: usize = 1 << 8;
+
+struct Timer(Instant);
+
+impl Timer {
+    fn new() -> Self {
+        Self(Instant::now())
+    }
+
+    fn step(&mut self, label: &str) {
+        println!("{}: {:?}", label, self.0.elapsed());
+        self.0 = Instant::now();
+    }
+}
+
+fn generate_leaves(count: usize) -> Vec<KeyPair> {
```

Comment:
I feels weird to add to the `MTree` API since it treats each leaf as an opaque vec of bytes, how about the addition of a `KeyCollection` type for these conveniences? (I was thinking about doing this in #112 anyways)

Something like this:

```rust
let keys = KeyCollection::new_ed25519(key_count)?;
let tree = keys.to_mtree(Hash::Blake2b256)?;
```

Reply:
Sure, that works. In the high-level implementation and Stronghold integration, we might be able to even make it one function. If you implement this in a different PR then consider this resolved.